x/poolmanager/routes_test.go

```diff
@@ -0,0 +1,446 @@
+		t.Errorf("Unexpected route: %+v", routes[0])
+	}
+}
+
+func TestFindTwoHopRoute(t *testing.T) {
+	routes := poolmanager.FindRoutes(graph, "token1", "token3", 2)
+
+	totalRoutes := 0
+	for _, subRoutes := range routes {
+		totalRoutes += len(subRoutes)
+	}
+
+	if totalRoutes != 2 {
+		t.Errorf("Expected 2 routes, got %d", totalRoutes)
+	}
+
+	if routes[0][0].PoolId != 1 || routes[0][0].Token != "token2" {
+		t.Errorf("Unexpected route: %+v", routes[0][0])
+	}
+
+	if routes[0][1].PoolId != 3 || routes[0][1].Token != "token3" {
+		t.Errorf("Unexpected route: %+v", routes[0][1])
+	}
+}
+
+func TestFindThreeHopRoute(t *testing.T) {
+	routes := poolmanager.FindRoutes(graph, "token1", "token4", 3)
+
+	totalRoutes := 0
+	for _, subRoutes := range routes {
+		totalRoutes += len(subRoutes)
+	}
+
+	if totalRoutes != 3 {
+		t.Errorf("Expected 3 routes, got %d", totalRoutes)
+	}
+
+	if routes[0][0].PoolId != 1 || routes[0][0].Token != "token2" {
+		t.Errorf("Unexpected route: %+v", routes[0][0])
+	}
+
+	if routes[0][1].PoolId != 3 || routes[0][1].Token != "token3" {
+		t.Errorf("Unexpected route: %+v", routes[0][1])
+	}
+
+	if routes[0][2].PoolId != 5 || routes[0][2].Token != "token4" {
+		t.Errorf("Unexpected route: %+v", routes[0][2])
+	}
+}
+
+func (s *KeeperTestSuite) TestGetSetDenomPairRoutes() {
+	// Set routes in state
+	routingGraph, err := s.App.PoolManagerKeeper.SetDenomPairRoutes(s.Ctx)
+	s.Require().NoError(err)
+	s.Require().Empty(routingGraph)
+
+	// Get routes from state and compare to expected
+	routingMap, err := s.App.PoolManagerKeeper.GetRouteMap(s.Ctx)
+	s.Require().NoError(err)
+	expectedRoutingMap := poolmanager.ConvertToMap(&routingGraph)
+	s.Require().Equal(expectedRoutingMap, routingMap)
+
+	// Prepare pools
+	s.PrepareBalancerPoolWithCoins(sdk.NewCoin("uosmo", sdk.NewInt(10000000000)), sdk.NewCoin("bar", sdk.NewInt(10000000000)))
+	s.PrepareConcentratedPoolWithCoins("uosmo", "foo")
+	s.PrepareCustomTransmuterPool(s.TestAccs[0], []string{"uosmo", "uion"})
+
+	// Create a pool that determines what the value of uosmo is in terms of USDC
+	poolManagerParams := s.App.PoolManagerKeeper.GetParams(s.Ctx)
+	poolManagerParams.MinValueForRoute.Denom = "usdc"
+	s.App.PoolManagerKeeper.SetParams(s.Ctx, poolManagerParams)
+	osmoUsdPool := s.PrepareConcentratedPoolWithCoins("uosmo", s.App.PoolManagerKeeper.GetParams(s.Ctx).MinValueForRoute.Denom)
+	s.CreateFullRangePosition(osmoUsdPool, sdk.NewCoins(sdk.NewCoin("uosmo", sdk.NewInt(10000000000)), sdk.NewCoin("usdc", sdk.NewInt(10000000000))))
+
+	// Set routes in state
+	routingGraph, err = s.App.PoolManagerKeeper.SetDenomPairRoutes(s.Ctx)
+	s.Require().NoError(err)
+	s.Require().NotEmpty(routingGraph)
+
+	// Get routes from state and compare to expected
+	routingMap, err = s.App.PoolManagerKeeper.GetRouteMap(s.Ctx)
+	s.Require().NoError(err)
+	expectedRoutingMap = poolmanager.ConvertToMap(&routingGraph)
+	s.Require().Equal(expectedRoutingMap, routingMap)
```

Comment:
Along with checking that the values from getter and setter are equal, can we also add check for the values in the graph as well?